File: Week 1/6.c
#include <stdio.h>
#include <stdbool.h>

bool isSorted(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        if (arr[i] < arr[i - 1]) {
            return false;
        }
    }
    return true;
}
int main() {
    int n;
    printf("Enter number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    if (isSorted(arr, n))
        printf("The array is sorted in non-decreasing order.\n");
    else
        printf("The array is NOT sorted in non-decreasing order.\n");

    return 0;
}

File: Week 1/6.py
class Solution:
    def isSorted(self, arr) -> bool:
        for i in range(1, len(arr)):
            if arr[i] < arr[i - 1]:
                return False
        return True
if __name__ == "__main__":
    arr = list(map(int, input("Enter array elements separated by spaces: ").split()))
    
    solution = Solution()
    result = solution.isSorted(arr)
    
    print("Is the array sorted in non-decreasing order?", result)
